plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/schema/SQLSchemaManager.java

```diff
@@ -82,29 +82,38 @@ public SQLSchemaManager(
     public void updateSchema(DBRProgressMonitor monitor) throws DBException {
         try {
             Connection dbCon = connectionProvider.getDatabaseConnection(monitor);
+            var autoCommit = dbCon.getAutoCommit();
+            dbCon.setAutoCommit(false);
             try (JDBCTransaction txn = new JDBCTransaction(dbCon)) {
-                int currentSchemaVersion = versionManager.getCurrentSchemaVersion(monitor, dbCon, targetDatabaseName);
-                // Do rollback in case some error happened during version check (makes sense for PG)
-                txn.rollback();
-                if (currentSchemaVersion < 0) {
-                    createNewSchema(monitor, dbCon);
-
-                    // Update schema version
-                    versionManager.updateCurrentSchemaVersion(monitor, dbCon, targetDatabaseName);
-                } else if (schemaVersionObsolete > 0 && currentSchemaVersion < schemaVersionObsolete) {
-                    dropSchema(monitor, dbCon);
-                    createNewSchema(monitor, dbCon);
-
-                    // Update schema version
-                    versionManager.updateCurrentSchemaVersion(monitor, dbCon, targetDatabaseName);
-                } else if (schemaVersionActual > currentSchemaVersion) {
-                    upgradeSchemaVersion(monitor, dbCon, currentSchemaVersion);
+                try {
```

Comment:
This nested `try` doesn't seem to be necessary? You can add `finally` to the `try-with-resources` block, too.

Reply:
The catch block from the nested `try` uses `txn`, which is not available from the `try-with-resource`'s `catch` block.